examples/auth/app/auth/components/LoginForm.tsx

```diff
@@ -1,69 +1,79 @@
 import React from "react"
 import {LabeledTextField} from "app/components/LabeledTextField"
-import {Form} from "react-final-form"
-import {FORM_ERROR} from "final-form"
+import {FormProvider, useForm} from "react-hook-form"
 import login from "app/auth/mutations/login"
 
 type LoginFormProps = {
   onSuccess?: () => void
 }
 
 export const LoginForm = (props: LoginFormProps) => {
+  const methods = useForm({
+    mode: "onBlur",
+    async resolver(values) {
+      const errors: {email?: string; password?: string} = {}
+      if (!values.email) errors.email = "Required"
+      if (!values.password) errors.password = "Required"
+      if (Object.keys(errors).length) {
+        return {values: {}, errors}
+      } else {
+        return {values, errors: {}}
+      }
+    },
+  })
+
   return (
     <div>
       <h1>Login</h1>
 
-      <Form
-        initialValues={{email: undefined, password: undefined}}
-        validate={(values) => {
-          const errors: {email?: string; password?: string} = {}
-          if (!values.email) errors.email = "Required"
-          if (!values.password) errors.password = "Required"
-          return errors
-        }}
-        onSubmit={async (values) => {
-          try {
-            await login({email: values.email, password: values.password})
-            props.onSuccess && props.onSuccess()
-          } catch (error) {
-            if (error.name === "AuthenticationError") {
-              return {[FORM_ERROR]: "Sorry, those credentials are invalid"}
-            } else {
-              return {
-                [FORM_ERROR]:
-                  "Sorry, we had an unexpected error. Please try again. - " + error.toString(),
+      <FormProvider {...methods}>
```

Comment:
I see you are following some of the examples in from the lib's own docs here, but this object does not contain only methods, so it's a bit of a misnomer, ihmo.

Reply:
Yeah I agree